core/src/main/java/hudson/util/ArgumentListBuilder.java

```diff
@@ -314,42 +314,46 @@ public String toStringWithQuote() {
      * @since 1.386
      */
     public ArgumentListBuilder toWindowsCommand(boolean escapeVars) {
-        StringBuilder quotedArgs = new StringBuilder();
+    	ArgumentListBuilder windowsCommand = new ArgumentListBuilder().add("cmd.exe", "/C");
         boolean quoted, percent;
-        for (String arg : args) {
+        for (int i = 0; i < args.size(); i++) {
+        	StringBuilder quotedArgs = new StringBuilder();
+        	String arg = args.get(i);
             quoted = percent = false;
-            for (int i = 0; i < arg.length(); i++) {
-                char c = arg.charAt(i);
+            for (int j = 0; j < arg.length(); j++) {
```

Comment:
Thanks @oleg-nenashev , can you suggest other name please? I always use i and j inside for in that order from upper to deeper.